File: shared/core/src/main/java/com/lacker/micros/utils/encrypt/AesUtils.java
package com.lacker.micros.utils.encrypt;

import org.springframework.security.crypto.codec.Hex;

import javax.crypto.BadPaddingException;
import javax.crypto.Cipher;
import javax.crypto.IllegalBlockSizeException;
import javax.crypto.KeyGenerator;
import javax.crypto.NoSuchPaddingException;
import javax.crypto.SecretKey;
import javax.crypto.spec.SecretKeySpec;
import java.nio.charset.StandardCharsets;
import java.security.InvalidKeyException;
import java.security.NoSuchAlgorithmException;
import java.security.SecureRandom;

public enum AesUtils {

    X;

    private static final String PASSWORD = "samples";
    private static final String KEY_ALGORITHM = "AES";
    private static final String SECURE_RANDOM_ALGORITHM = "SHA1PRNG";
    private static final String DEFAULT_CIPHER_ALGORITHM = "AES/ECB/PKCS5Padding";

    public String encrypt(String content) {
        try {
            Cipher cipher = Cipher.getInstance(DEFAULT_CIPHER_ALGORITHM);
            cipher.init(Cipher.ENCRYPT_MODE, provideSecretKey());
            return new String(Hex.encode(cipher.doFinal(content.getBytes(StandardCharsets.UTF_8))));
        } catch (NoSuchAlgorithmException | NoSuchPaddingException | BadPaddingException | IllegalBlockSizeException | InvalidKeyException ex) {
            throw new IllegalArgumentException(ex);
        }
    }

    public String decrypt(String content) {
        try {
            Cipher cipher = Cipher.getInstance(DEFAULT_CIPHER_ALGORITHM);
            cipher.init(Cipher.DECRYPT_MODE, provideSecretKey());
            return new String(cipher.doFinal(Hex.decode(content)), StandardCharsets.UTF_8);
        } catch (NoSuchAlgorithmException | NoSuchPaddingException | BadPaddingException | IllegalBlockSizeException | InvalidKeyException ex) {
            throw new IllegalArgumentException(ex);
        }
    }

    private SecretKey provideSecretKey() throws NoSuchAlgorithmException {
        KeyGenerator keyGen = KeyGenerator.getInstance(KEY_ALGORITHM);
        SecureRandom secureRandom = SecureRandom.getInstance(SECURE_RANDOM_ALGORITHM);
        secureRandom.setSeed(PASSWORD.getBytes(StandardCharsets.UTF_8));
        keyGen.init(128, secureRandom);
        return new SecretKeySpec(keyGen.generateKey().getEncoded(), KEY_ALGORITHM);
    }
}

File: library/core/src/main/kotlin/com/mlacker/micros/utils/encrypt/AesUtils.kt
package com.mlacker.micros.utils.encrypt

import org.springframework.security.crypto.codec.Hex
import java.nio.charset.StandardCharsets
import java.security.InvalidKeyException
import java.security.NoSuchAlgorithmException
import java.security.SecureRandom
import javax.crypto.*
import javax.crypto.spec.SecretKeySpec

enum class AesUtils {

    X;

    fun encrypt(content: String): String {
        return try {
            val cipher = Cipher.getInstance(DEFAULT_CIPHER_ALGORITHM)
            cipher.init(Cipher.ENCRYPT_MODE, provideSecretKey())
            String(Hex.encode(cipher.doFinal(content.toByteArray(StandardCharsets.UTF_8))))
        } catch (ex: NoSuchAlgorithmException) {
            throw IllegalArgumentException(ex)
        } catch (ex: NoSuchPaddingException) {
            throw IllegalArgumentException(ex)
        } catch (ex: BadPaddingException) {
            throw IllegalArgumentException(ex)
        } catch (ex: IllegalBlockSizeException) {
            throw IllegalArgumentException(ex)
        } catch (ex: InvalidKeyException) {
            throw IllegalArgumentException(ex)
        }
    }

    fun decrypt(content: String): String {
        return try {
            val cipher = Cipher.getInstance(DEFAULT_CIPHER_ALGORITHM)
            cipher.init(Cipher.DECRYPT_MODE, provideSecretKey())
            String(cipher.doFinal(Hex.decode(content)), StandardCharsets.UTF_8)
        } catch (ex: NoSuchAlgorithmException) {
            throw IllegalArgumentException(ex)
        } catch (ex: NoSuchPaddingException) {
            throw IllegalArgumentException(ex)
        } catch (ex: BadPaddingException) {
            throw IllegalArgumentException(ex)
        } catch (ex: IllegalBlockSizeException) {
            throw IllegalArgumentException(ex)
        } catch (ex: InvalidKeyException) {
            throw IllegalArgumentException(ex)
        }
    }

    private fun provideSecretKey(): SecretKey {
        val keyGen = KeyGenerator.getInstance(KEY_ALGORITHM)
        val secureRandom = SecureRandom.getInstance(SECURE_RANDOM_ALGORITHM)
        secureRandom.setSeed(PASSWORD.toByteArray(StandardCharsets.UTF_8))
        keyGen.init(128, secureRandom)
        return SecretKeySpec(keyGen.generateKey().encoded, KEY_ALGORITHM)
    }

    companion object {
        private const val PASSWORD = "samples"
        private const val KEY_ALGORITHM = "AES"
        private const val SECURE_RANDOM_ALGORITHM = "SHA1PRNG"
        private const val DEFAULT_CIPHER_ALGORITHM = "AES/ECB/PKCS5Padding"
    }
}